sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala

```diff
@@ -285,334 +290,61 @@ class TableIdentifierParserSuite extends SparkFunSuite with SQLHelper {
+          sys.env.contains("SPARK_HOME"), "spark.test.home or SPARK_HOME is not set.")
+        sys.props.getOrElse("spark.test.home", sys.env("SPARK_HOME"))
+      }
+      java.nio.file.Paths.get(sparkHome, "sql", "catalyst", "src", "main", "antlr4", "org",
+        "apache", "spark", "sql", "catalyst", "parser", "SqlBase.g4").toFile
+    }
+    fileToString(sqlBasePath).split("\n")
+  }
 
-  val reservedKeywordsInAnsiMode = Set(
-    "all",
-    "and",
-    "anti",
-    "any",
-    "as",
-    "authorization",
-    "both",
-    "case",
-    "cast",
-    "check",
-    "collate",
-    "column",
-    "constraint",
-    "create",
-    "cross",
-    "current_date",
-    "current_time",
-    "current_timestamp",
-    "current_user",
-    "day",
-    "distinct",
-    "else",
-    "end",
-    "escape",
-    "except",
-    "false",
-    "fetch",
-    "for",
-    "foreign",
-    "from",
-    "full",
-    "grant",
-    "group",
-    "having",
-    "hour",
-    "in",
-    "inner",
-    "intersect",
-    "into",
-    "join",
-    "is",
-    "leading",
-    "left",
-    "minute",
-    "month",
-    "natural",
-    "not",
-    "null",
-    "on",
-    "only",
-    "or",
-    "order",
-    "outer",
-    "overlaps",
-    "primary",
-    "references",
-    "right",
-    "select",
-    "semi",
-    "session_user",
-    "minus",
-    "second",
-    "some",
-    "table",
-    "then",
-    "to",
-    "trailing",
-    "union",
-    "unique",
-    "unknown",
-    "user",
-    "using",
-    "when",
-    "where",
-    "with",
-    "year")
+  private def parseAntlrGrammars(startTag: String, endTag: String)
+      (f: PartialFunction[String, Option[String]]): Set[String] = {
+    val keywords = new mutable.ArrayBuffer[String]
+    val default = (_: String) => None
+    var start = false
+    for (line <- sqlSyntaxDefs) {
+      if (line.trim.startsWith(startTag)) {
+        start = true
+      } else if (line.trim.startsWith(endTag)) {
+        start = false
+      } else if (start) {
+        f.applyOrElse(line, default).foreach(keywords += _)
+      }
+    }
+    keywords.map(_.trim.toLowerCase(Locale.ROOT)).toSet
+  }
 
-  val nonReservedKeywordsInAnsiMode = allCandidateKeywords -- reservedKeywordsInAnsiMode
+  private def parseSyntax(startTag: String, endTag: String): Set[String] = {
+    val kwDef = """\s*[\|:]\s*([A-Z_]+)\s*""".r
+    parseAntlrGrammars(startTag, endTag) {
+      // Parses a pattern, e.g., `    | AFTER`
+      case kwDef(literal) => Some(literal)
+    }
+  }
+
+  // All the SQL keywords defined in `SqlBase.g4`
+  val allCandidateKeywords = {
+    val kwDef = """([A-Z_]+):.+;""".r
+    parseAntlrGrammars("//--SPARK-KEYWORD-LIST-START", "//--SPARK-KEYWORD-LIST-END") {
+      // Parses a pattern, e.g., `RLIKE: 'RLIKE' | 'REGEXP';`
+      case kwDef(literal) => Some(literal)
+    }
+  }
+
+  val nonReservedKeywordsInAnsiMode = parseSyntax(
+    "//--ANSI-NON-RESERVED-START", "//--ANSI-NON-RESERVED-END")
+
+  val reservedKeywordsInAnsiMode = {
```

Comment:
ansi reserved keywords are "full keywords -- non-reserved keywords".

Reply:
"non-reserved keywords" means  non-reserved keywords in ansi and strict non-reserved keywords?

Does previous `allCandidateKeywords` include strict non-reserved keywords?

Reply:
okay, fixed.